Modules/_decimal/_decimal.c

```diff
@@ -1895,13 +1935,23 @@ PyDecType_New(PyTypeObject *type)
 
     return (PyObject *)dec;
 }
-#define dec_alloc() PyDecType_New(&PyDec_Type)
+#define dec_alloc(st) PyDecType_New(st->PyDec_Type)
```

Comment:
Always use good macro hygiene :)

```suggestion
#define dec_alloc(st) PyDecType_New((st)->PyDec_Type)
```